### PR TITLE
2nd try fix Issue 15579 - extern(C++) interfaces/multiple-inheritance

### PR DESCRIPTION
**Pull request by WalterBright**
This should fix the Win64 problems.

https://issues.dlang.org/show_bug.cgi?id=15579

### CODE REVIEW

File: src/expression.d [hunk 1]
```diff
@@ -9257,6 +9271,7 @@ public:
                         return new ErrorExp();
                     ue = cast(UnaExp)e1;
                 }
+//<<>>
```
Comment (ibuclaw) [inline src/expression.d]: What's this for?

File: src/expression.d [hunk 2]
```diff
@@ -9268,8 +9283,10 @@ public:
                 // See if we need to adjust the 'this' pointer
                 AggregateDeclaration ad = f.isThis();
                 ClassDeclaration cd = ue.e1.type.isClassHandle();
+//printf("test1 %s\n", f.toChars());
                 if (ad && cd && ad.isClassDeclaration())
                 {
+//printf("test2 %s ad %s cd %s i %p\n", f.toChars(), ad.toChars(), cd.toChars(), f.interfaceVirtual);
```
Comment (ibuclaw) [inline src/expression.d]: Comments should be aligned with the rest of the code or removed.

**Comment by WalterBright** [comment]
Much more thorough testing. Crossing fingers!

**Comment by TurkeyMan** [comment]
Oh, this was not pulled...
Does this address the new issue I posted of `this` pointer adjustment?

**Comment by WalterBright** [comment]
Yes - you'll see it in the attached test case. And note that this is a peculiarity of VC++.

**Comment by TurkeyMan** [comment]
Can someone merge this?
I've been trying to build DMD all day, and failed, as usual.

**Comment by yebblies** [comment]
Try building with dmc.

**Comment by andralex** [comment]
Thx @WalterBright ! Also thx to @TurkeyMan for all detective work!

**Pull request merged**